staging/src/k8s.io/kubectl/pkg/cmd/apiresources/apiresources.go

```diff
@@ -205,18 +228,36 @@ func (o *APIResourceOptions) RunAPIResources() error {
 				APIGroupVersion: gv.String(),
 				APIResource:     resource,
 			})
+			apiResources = append(apiResources, resource)
+		}
+		if len(apiResources) > 0 {
+			apiList.APIResources = apiResources
+			filteredList = append(filteredList, apiList)
 		}
 	}
 
-	if o.NoHeaders == false && o.Output != "name" {
-		if err = printContextHeaders(w, o.Output); err != nil {
+	if o.NoHeaders == false && *o.PrintFlags.OutputFormat == "" || (*o.PrintFlags.OutputFormat == "wide" && !o.NoHeaders) {
```

Comment:
```suggestion
	if o.NoHeaders == false && (*o.PrintFlags.OutputFormat == "" || *o.PrintFlags.OutputFormat == "wide") {
```

Reply:
Thanks. My local branch had the `!o.NoHeaders` part of it already, but not the correct OR check that you provided. Done.